metricbeat/module/kubernetes/kubecontroller/kubecontroller.go

```diff
@@ -0,0 +1,67 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package kubecontroller
+
+import (
+	"github.com/elastic/beats/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+func init() {
+	mapping := &prometheus.MetricsMapping{
+		Metrics: map[string]prometheus.MetricMap{
+			// Process metrics
+			"process_cpu_seconds_total":     prometheus.Metric("process.cpu.sec"),
+			"process_resident_memory_bytes": prometheus.Metric("process.memory.resident.bytes"),
+			"process_virtual_memory_bytes":  prometheus.Metric("process.memory.virtual.bytes"),
+			"process_open_fds":              prometheus.Metric("process.fds.open.count"),
+			"process_start_time_seconds":    prometheus.Metric("process.started.sec"),
+
+			// HTTP server metrics
+			"http_request_duration_microseconds": prometheus.Metric("http.request.duration.us"),
+			"http_request_size_bytes":            prometheus.Metric("http.request.size.bytes"),
+			"http_response_size_bytes":           prometheus.Metric("http.response.size.bytes"),
+			"http_requests_total":                prometheus.Metric("http.request.count"),
+
+			// REST metrics
+			"rest_client_requests_total": prometheus.Metric("client.request.count"),
+
+			// Controller metrics
+			"workqueue_longest_running_processor_seconds": prometheus.Metric("workqueue.longestrunning.sec"),
+			"workqueue_unfinished_work_seconds":           prometheus.Metric("workqueue.unfinished.sec"),
```

Comment:
Should we get other workqueue metrics as `workqueue_depth`, or `adds_total`?
And the ones that are histograms could be interesting too, are we intentionally leaving these by now?

Reply:
please, enum any metrics you think are useful and I'll add it.
not adding all metrics is intentional mostly because of saving space or not being able to get useful visualisations later on. If you are missing some metrics, I'll add them.

Reply:
Not sure which ones I'd collect, but I think that at least the gauges and counter ones, that should be cheap as we already collect some of them, that have the same labels. If I am not missing anyone they would be: `workqueue_adds_total`, `workqueue_depth`, `workqueue_retries_total`,